src/osd/osd_op_util.cc

```diff
@@ -258,6 +283,10 @@ int OpInfo::set_from_op(
   if (rmw_flags == 0)
     return -EINVAL;
 
+  if (read_op_count != stat_op_count || may_read_data) {
```

Comment:
A question/concern related to seemingly unneeded variables above: in which case would `read_op_count != stat_op_count` be true, resulting in `set_may_read_data` method getting called despite `may_read_data` being false?